signet-program/programs/signet/src/lib.rs

```diff
@@ -1,18 +1,190 @@
-//! Sig.Network signing program for accepting signature requests and providing responses from the Sig.Network.
+//! # Signet Chain Signatures Program
```

Comment:
This module level doc looks good, but I would recommend to have this and other flows well documented in our docs and referenced from all other places. Having it in multiple places makes it hard to maintain.